frontend/lib/pattern-helpers.spec.js

```diff
@@ -1,10 +1,28 @@
 import { test, given } from 'sazerac'
-import { patternToOptions } from './pattern-helpers'
+import { patternToOptions, removeRegexpFromPattern } from './pattern-helpers'
 
 describe('Badge URL functions', function() {
   test(patternToOptions, () => {
     given('[^\\/]+?').expect(undefined)
     given('abc|[^\\/]+').expect(undefined)
     given('abc|def|ghi').expect(['abc', 'def', 'ghi'])
   })
+
+  test(removeRegexpFromPattern, () => {
+    given('/appveyor/ci/:user/:repo').expect('/appveyor/ci/:user/:repo')
+    given('/discourse/:scheme(http|https)/:host/topics').expect(
+      '/discourse/:scheme/:host/topics'
```

Comment:
This is a nice enhancement. Previously routes like `:variant(maintainability|maintainability-percentage|tech-debt|issues)` were generating some really long URLs on the index page and we don't need to do that now that we have drop-downs in the modal.